cluster/points_writer.go

```diff
@@ -196,6 +202,13 @@ func (w *PointsWriter) WritePoints(database, retentionPolicy string, consistency
 	w.statMap.Add(statWriteReq, 1)
 	w.statMap.Add(statPointWriteReq, int64(len(points)))
 
+	if w.MaxConcurrentWrites > 0 {
```

Comment:
The reason why I did it this way was to avoid putting this into the http handler. We currently have the query limit inside of the query executor so I thought to put this in the points writer.

I think it would be good to refactor the points writer to read from a stream though so it would be easier to separate out this functionality into the points writer. Putting it in the points writer also allows us to restrict the number of maximum concurrent writers from other sources such as udp, collectd, graphite, and continuous queries rather than only through the httpd service.